README.md
=== TheUx Slider ===

Tags: slider, parallax, slideshow   
Requires at least: 3.5.1   
Tested up to: 3.6   
Stable tag: 0.9.9  
License: GPLv2 or later
License URI: http://www.gnu.org/licenses/gpl-2.0.html

TheUx Slider is a real world responsive, unbranded, premium and open source slider.

== Description ==

TheUx Slider is a premium slider with a twist. No tricks, no lite version, no ads, no donation buttons, no backlinks and no brand. A clean and robust premium slider that is 100% open source and full of features that are only found on those paid sliders. Pay no more!

= Features =

*   TheUx Series.
*   100% Responsive.
*   Animate multiple elements.
*   Full control over each element.
*   Unlimited elements.
*   Unlimited slides.
*   Group elements in steps.
*   Slideshows! Organize your slides into multiple slideshows.
*   Easy theme integration.
*   No fancy UI.
*   Low learning curve.
*   Drag-and-drop image uploading.
*   No settings page!
*   OOP approach.
*   Singleton free!!
*   and much more!


= TheUx Serie =

TheUx is a premium open source series of WordPress plugins. More than a handy crafted code, our signature is to be an unbranded plugin. No lite version, no ads, no donation buttons, no backlinks and no fancy UI. TheUx series use the standard WordPress UI and coding convention. It's a real premium plugin for free.


== Installation ==

1. Upload `theux-slider` directory to the `/wp-content/plugins/` directory
1. Activate the plugin through the 'Plugins' menu in WordPress

= Adding TheUx Slider =

It's very easy to include the slider on your theme. Just use the following shortcode on any page/post.

    [slider]

Playing with the parameters:

    [slider dimensions="900,300" autochange=true pauseonhover=true]


To use shortcode within a PHP file (outside the post editor).
     
     <?php echo do_shortcode('[slider]'); ?>


**Available parameters and their default values**

    slideshow: '',
    controls: true,
    pager: false,
    autochange: true,
    pauseonhover: false,
    slideendanimation: false
    backgroundanimation: false
    backgroundx: 500
    backgroundy: 500
    backgroundspeed: 2500
    backgroundease: 'easeOutCubic'   
    fullwidth: true,
    responsive: true,
    dimensions: '',
    increase: false,
      
   
    
== Screenshots ==

1. Slides listing page.
2. Slide elements page.
3. Slideshow settings.



== Frequently Asked Questions ==

Please go ahead and read this FAQ to understand more about TheUX Slider philosophy.


= What is the TheUx Slider goal?

The TUS (TheUx Slider) goal is to be the main slider of your website and not jack of 
all trades slider. 


= Why doesn't TUS have a settings page?

TUS was designed with the Single Responsibility Principle in mind. And for that, you 
don't need that many options. All the options you need you to SET UP the slider you
pass on shortcode. The options used on the SLIDES are defined when you add or edit them.


= But if I need more sliders, will I need to install a second slider pluing for that? =

This answer will sound a bit weird, but stay with me. 

Really? Do you really need to have more than one slider per website? Yes, I wrote website and not 
page. Please don't make that mistake. By nature, most of sliders sucks [Sliders Suck](http://krogsgard.com/2013/sliders-suck)

= WTF??? =

Let me put this way. Sliders are fancy animations that most of the time are there just a distract the 
user, and not to make any real convertion. As any UI element on your website, the slider must have a 
reason to exist. Putting it there just for esthetic purpose may hurt your website convertion and distracting
the visitors for complete your goals. The rule of thumb is: if the slider is part of your website strategie
and its serve a specific goal, no worries. Go for it! Otherwise, I strongly suggest you to rethink its use. 

= Ok, I'm sold. Just out of curiosity, why did you write this plugin then? =

Come on..it's fun. Don't agree? 

1. Learning is my main reason. TheUx Slider is real plugin. It has OOP, PHP, JS, WP plugins development..and so on.
All the things you need to improve your skills. 

1. I want people to be aware about the this slider fever. It's unbelievable that people are paying for 'premium' sliders.
Not only that, it's a common practice for 'premium themes' to include two, three or even four different sliders. I don't 
know if it's only me, but that is nonsense.


== Acknowledges ==

 This plugin only exists because of some amazing people out there. A very special thanks to:

* Gijs Jorissen - [Wordpress-Cuztom-Helper](http://https://github.com/Gizburdt/Wordpress-Cuztom-Helper)
* Mario Jäckle - [FractionSlider](http://http://jacksbox.de/stuff/jquery-fractionslider/)


== Change Log ==

= 0.9.9 (10 September 2013) =

* Beta release.


== License ==
	
	Copyright 2013 Daniel Zilli

    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License, version 2, as 
    published by the Free Software Foundation.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with this program; if not, write to the Free Software
    Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301 USA


ChangeLog.md
## 0.9.9 (10 September 2013)

* Beta release.


includes/admin.php
<?php
/**
 * Administration.
 *
 * @package   TUS
 * @author    Daniel Zilli
 * @version   1.0.0
 */

/**
 * Register the custom post type.
 *
 * @since	1.0.0
 */
$tus  = new Cuztom_Post_Type( 'theux_slider', 
	array(
		'menu_icon'  	=> plugins_url('assets/images/menu-icon.png',  dirname(__FILE__) ),
	    'supports' 		=> array( 'title' ),
	),
	array(
	    'name'              => _x( 'Slider', 'tus' ),
	    'singular_name'     => _x( 'Slider', 'tus' ),
	    'search_items'      => __( 'Search Slides' ),
	    'all_items'         => __( 'All Slides' ),
	    'parent_item'       => __( 'Parent Slide' ),
	    'parent_item_colon' => __( 'Parent Slides:' ),
	    'edit_item'         => __( 'Edit Slide' ),
	    'update_item'       => __( 'Update Slide' ),
	    'add_new_item'      => __( 'Add New Slide' ),
	    'new_item_name'     => __( 'New Slide Name' ),
	    'menu_name'         => __( 'Slider' ),
	) 
);


/**
 * Register taxonomy.
 *
 * @since	1.0.0
 */
$slideshow = register_cuztom_taxonomy( 'Slideshow', 'theux_slider', 
        array(
            'show_admin_column'     => TRUE,
            'admin_column_sortable' => TRUE,
            'admin_column_filter'   => TRUE,
            'hierarchical' 			=> FALSE,
        ));

/**
 * Register the meta-boxes.
 *
 * @since	1.0.0
 */


/* Background options */
$tus->add_meta_box( 
	'tus_background_id',
	'Background Slide Options', 
	array(
	        array(
	            'name'          => 'bg_image',
	            'label'         => __( 'Image', 'tus' ),
	            'description'   => __( 'Select the background image for the slide.', 'tus' ),
	            'type'          => 'image'
	        ),
	        array(
	            'name'          => 'bg_color',
	            'label'         => __( 'Color', 'tus' ),
	            'description'   => __( 'Select the background color.', 'tus' ),
	            'type'          => 'color'
	        ),
	    ),
	'normal', // context
	'high' // priority
);

/* Bundle metabox */
$tus->add_meta_box( 
	'tus_elements_id',
	'Slide Element Options', 
	array(
	        'bundle', 
	    array(
	        array(
	            'name'          => 'element_image',
	            'label'         => __( 'Image', 'tus' ),
	            'description'   => __( 'Select an image', 'tus' ),
	            'type'          => 'image'
	        ),
			array(
				'name'          => 'element_video',
				'label'         => __( 'Video Link', 'tus' ),
				'description'   => __( 'Enter your video link', 'tus' ),
				'type'          => 'text'
			),
			array(
				'name'          => 'element_video_dimension',
				'label'         => __( 'Video Dimensions', 'tus' ),
				'description'   => __( 'Enter element dimension as width,height.<br />Example: 500,300', 'tus' ),
				'type'          => 'text'
			),
	        array(
	            'name'          => 'element_caption',
	            'label'         => __( 'Caption', 'tus' ),
	            'description'   => __( 'Enter your caption', 'tus' ),
	            'type'          => 'textarea'
	        ),
	        array(
		        'name'          => 'element_caption_theme',
		        'label'         => __( 'Caption Themes', 'tus' ),
		        'description'   => __( 'Choose the color scheme for the captions.', 'tus'),
		        'type'          => 'select',
		        'options'       => array(
		        	'none'			=>  __( 'None', 'tus'),
	            	'transparent'	=>	__( 'transparent', 'tus'),		        	
	            	'white'			=>	__( 'white', 'tus'),
	            	'black'			=>	__( 'black', 'tus'),
	            	'light-green'	=>	__( 'light-green', 'tus'),
	            	'green'			=>	__( 'green', 'tus'),
	            	'orange'		=>  __( 'orange', 'tus'),
	            	'turky'			=>	__( 'turky', 'tus'),
	            	'blue'			=>	__( 'blue', 'tus'),
	            	'light-red'		=>	__( 'light-red', 'tus'),
	            	'brown'			=>	__( 'brown', 'tus'),
		        ),
		        'default_value' => 'none',
		    ),
	        array(
	            'name'          => 'element_option_position',
	            'label'         => __( 'Postion', 'tus'),
	            'description'   => __( 'Enter element position as Y,X.<br>Example: 100,10', 'tus'),
	            'type'          => 'text'
	        ),
	        array(
	            'name'          => 'element_option_delay',
	            'label'         => __( 'Delay', 'tus'),
	            'description'   => __( 'Time in ms before the <b>in</b> transition starts.', 'tus'),
	            'type'          => 'text'
	        ),
	        array(
	            'name'          => 'element_option_time',
	            'label'         => __( 'Time', 'tus'),
	            'description'   => __( 'Time after which the elements animation is complete in ms.', 'tus'),
	            'type'          => 'text'
	        ),
	        array(
		        'name'          => 'element_option_in',
		        'label'         => __( 'Data In', 'tus'),
		        'description'   => __( 'Type of the in-animation.', 'tus'),
		        'type'          => 'select',
		        'options'       => array(
		        	'none'			=>  __( 'None', 'tus'),        	
	            	'fade'			=>	__( 'fade', 'tus'),
	            	'left'			=>	__( 'left', 'tus'),
	            	'topLeft'		=>	__( 'topLeft', 'tus'),
	            	'bottomLeft'	=>	__( 'bottomLeft', 'tus'),
	            	'right'			=>  __( 'right', 'tus'),
	            	'topRight'		=>	__( 'topRight', 'tus'),
	            	'bottomRight'	=>	__( 'bottomRight', 'tus'),
	            	'top'			=>	__( 'top', 'tus'),
	            	'bottom'		=>	__( 'bottom', 'tus'),
		        ),
		        'default_value' => 'left',
		    ),
	        array(
		        'name'          => 'element_option_out',
		        'label'         => __( 'Data Out', 'tus'),
		        'description'   => __( 'Type of the out-animation.', 'tus'),
		        'type'          => 'select',
		        'options'       => array(
		        	'none'			=>  __( 'None', 'tus'),		        	
	            	'fade'			=>	__( 'fade', 'tus'),
	            	'left'			=>	__( 'left', 'tus'),
	            	'topLeft'		=>	__( 'topLeft', 'tus'),
	            	'bottomLeft'	=>	__( 'bottomLeft', 'tus'),
	            	'right'			=>  __( 'right', 'tus'),
	            	'topRight'		=>	__( 'topRight', 'tus'),
	            	'bottomRight'	=>	__( 'bottomRight', 'tus'),
	            	'top'			=>	__( 'top', 'tus'),
	            	'bottom'		=>	__( 'bottom', 'tus'),
		        ),
		        'default_value' => 'left',
		    ),
	        array(
		        'name'          => 'element_option_step',
		        'label'         => __( 'Step', 'tus'),
		        'description'   => __( 'Group elements in different steps. Elements of the next step will not start before the previous step is finished.', 'tus'),
		        'type'          => 'select',
		        'options'       => array(
		        	'none' 	=> __( 'none', 'tus'),
	            	'1'		=>	'1',
	            	'2'		=>	'2',
	            	'3'		=>	'3',
	            	'4'		=>	'4',
	            	'5'		=>  '5',
	            	'6'		=>	'6',
	            	'7'		=>	'7',
	            	'8'		=>	'8',
	            	'9'		=>	'9',
	            	'10'	=>	'10',
	            	'11'	=>	'11',
	            	'12'	=>	'12',
	            	'13'	=>	'13',
	            	'14'	=>  '14',
	            	'15'	=>	'15',
	            	'16'	=>	'16',
	            	'17'	=>	'17',
	            	'18'	=>	'18',
	            	'19'	=>	'19',
	            	'20'	=>	'20',
		        ),
		        'default_value' => 'none',
		    ),
	        array(
		        'name'          => 'element_option_ease_in',
		        'label'         => __( 'Ease In', 'tus'),
		        'description'   => __( 'Easing for the in-animations.', 'tus'),
		        'type'          => 'select',
		        'options'       =>  array(
		        	'none' 				=> __( 'none', 'tus'),
	        		'easeInQuad' 		=> __( 'easeInQuad', 'tus'),
					'easeOutQuad' 		=> __( 'easeOutQuad', 'tus'),
					'easeInOutQuad' 	=> __( 'easeInOutQuad', 'tus'),
					'easeInCubic' 		=> __( 'easeInCubic', 'tus'),
					'easeOutCubic' 		=> __( 'easeOutCubic', 'tus'),
					'easeInOutCubic' 	=> __( 'easeInOutCubic', 'tus'),
					'easeInQuart' 		=> __( 'easeInQuart', 'tus'),
					'easeOutQuart' 		=> __( 'easeOutQuart', 'tus'),
					'easeInOutQuart' 	=> __( 'easeInOutQuart', 'tus'),
					'easeInQuint' 		=> __( 'easeInQuint', 'tus'),
					'easeOutQuint' 		=> __( 'easeOutQuint', 'tus'),
					'easeInOutQuint' 	=> __( 'easeInOutQuint', 'tus'),
					'easeInSine' 		=> __( 'easeInSine', 'tus'),
					'easeOutSine' 		=> __( 'easeOutSine', 'tus'),
					'easeInOutSine' 	=> __( 'easeInOutSine', 'tus'),
					'easeInExpo' 		=> __( 'easeInExpo', 'tus'),
					'easeOutExpo' 		=> __( 'easeOutExpo', 'tus'),
					'easeInOutExpo' 	=> __( 'easeInOutExpo', 'tus'),
					'easeInCirc' 		=> __( 'easeInCirc', 'tus'),
					'easeOutCirc' 		=> __( 'easeOutCirc', 'tus'),
					'easeInOutCirc' 	=> __( 'easeInOutCirc', 'tus'),
					'easeInElastic' 	=> __( 'easeInElastic', 'tus'),
					'easeOutElastic' 	=> __( 'easeOutElastic', 'tus'),
					'easeInOutElastic' 	=> __( 'easeInOutElastic', 'tus'),
					'easeInBack' 		=> __( 'easeInBack', 'tus'),
					'easeOutBack' 		=> __( 'easeOutBack', 'tus'),
					'easeInOutBack' 	=> __( 'easeInOutBack', 'tus'),
					'easeInBounce' 		=> __( 'easeInBounce', 'tus'),
					'easeOutBounce' 	=> __( 'easeOutBounce', 'tus'),
					'easeInOutBounce' 	=> __( 'easeInOutBounce', 'tus'),
					),
		        'default_value' => 'easeInQuad',
		    ),

	        array(
		        'name'          => 'element_option_ease_out',
		        'label'         => __( 'Ease Out', 'tus'),
		        'description'   => __( 'Easing for the out-animations.', 'tus'),
		        'type'          => 'select',
			 'options'       =>  array(
		        	'none' 				=> __( 'none', 'tus'),
	        		'easeInQuad' 		=> __( 'easeInQuad', 'tus'),
					'easeOutQuad' 		=> __( 'easeOutQuad', 'tus'),
					'easeInOutQuad' 	=> __( 'easeInOutQuad', 'tus'),
					'easeInCubic' 		=> __( 'easeInCubic', 'tus'),
					'easeOutCubic' 		=> __( 'easeOutCubic', 'tus'),
					'easeInOutCubic' 	=> __( 'easeInOutCubic', 'tus'),
					'easeInQuart' 		=> __( 'easeInQuart', 'tus'),
					'easeOutQuart' 		=> __( 'easeOutQuart', 'tus'),
					'easeInOutQuart' 	=> __( 'easeInOutQuart', 'tus'),
					'easeInQuint' 		=> __( 'easeInQuint', 'tus'),
					'easeOutQuint' 		=> __( 'easeOutQuint', 'tus'),
					'easeInOutQuint' 	=> __( 'easeInOutQuint', 'tus'),
					'easeInSine' 		=> __( 'easeInSine', 'tus'),
					'easeOutSine' 		=> __( 'easeOutSine', 'tus'),
					'easeInOutSine' 	=> __( 'easeInOutSine', 'tus'),
					'easeInExpo' 		=> __( 'easeInExpo', 'tus'),
					'easeOutExpo' 		=> __( 'easeOutExpo', 'tus'),
					'easeInOutExpo' 	=> __( 'easeInOutExpo', 'tus'),
					'easeInCirc' 		=> __( 'easeInCirc', 'tus'),
					'easeOutCirc' 		=> __( 'easeOutCirc', 'tus'),
					'easeInOutCirc' 	=> __( 'easeInOutCirc', 'tus'),
					'easeInElastic' 	=> __( 'easeInElastic', 'tus'),
					'easeOutElastic' 	=> __( 'easeOutElastic', 'tus'),
					'easeInOutElastic' 	=> __( 'easeInOutElastic', 'tus'),
					'easeInBack' 		=> __( 'easeInBack', 'tus'),
					'easeOutBack' 		=> __( 'easeOutBack', 'tus'),
					'easeInOutBack' 	=> __( 'easeInOutBack', 'tus'),
					'easeInBounce' 		=> __( 'easeInBounce', 'tus'),
					'easeOutBounce' 	=> __( 'easeOutBounce', 'tus'),
					'easeInOutBounce' 	=> __( 'easeInOutBounce', 'tus'),
					),
		        'default_value' => 'easeOutQuad',
		    ),

	    )
	)
);


/**
 * Display custom messages.
 *
 * @since	1.0.0
 */
function tus_updated_messages( $messages ) {

	global $post, $post_ID;
	$messages['theux_slider'] = array (
		0  => '',
		1  => sprintf( __( 'Slide updated. <a href="%s">View slide</a>', 'tus' ), esc_url( get_permalink($post_ID) ) ),
		2  => __( 'Custom field updated.', 'tus' ),
		3  => __( 'Custom field deleted.', 'tus' ),
		4  => __( 'Slide updated.', 'tus' ),
		5  => isset($_GET['revision']) ? sprintf( __( 'Slide restored to revision from %s', 'tus' ), wp_post_revision_title( (int) $_GET['revision'], FALSE ) ) : FALSE,
		6  => sprintf( __( 'Slide published. <a href="%s">View slide</a>', 'tus' ), esc_url( get_permalink($post_ID) ) ),
		7  => __( 'Slide saved.', 'tus' ),
		8  => sprintf( __( 'Slide submitted. <a target="_blank" href="%s">Preview slide</a>', 'tus' ), esc_url( add_query_arg( 'preview', 'true', get_permalink($post_ID) ) ) ),
		9  => sprintf( __( 'Slide scheduled for: <strong>%1$s</strong>. <a target="_blank" href="%2$s">Preview slide</a>', 'tus' ), date_i18n( __( 'M j, Y @ G:i', 'tus' ), strtotime( $post->post_date ) ), esc_url( get_permalink($post_ID) ) ),
		10 => sprintf( __( 'Slide draft updated. <a target="_blank" href="%s">Preview slide</a>', 'tus' ), esc_url( add_query_arg( 'preview', 'true', get_permalink($post_ID) ) ) ) );

	return $messages;
}
add_filter( 'post_updated_messages', 'tus_updated_messages' );


/**
 * Parse Video.
 *
 * @author  takien, slaffko
 * @since	1.0.0
 */
function tus_video_url($url,$return='embed',$width='',$height='', $element_position='', $element_step='',$rel=0){
    $urls = parse_url($url);

    //url is http://vimeo.com/xxxx
    if($urls['host'] == 'vimeo.com'){
        $vid = ltrim($urls['path'],'/');
    }
    //url is http://youtu.be/xxxx
    else if($urls['host'] == 'youtu.be'){
        $yid = ltrim($urls['path'],'/');
    }
    //url is http://www.youtube.com/embed/xxxx
    else if(strpos($urls['path'],'embed') == 1){
        $yid = end(explode('/',$urls['path']));
    }
     //url is xxxx only
    else if(strpos($url,'/')===false){
        $yid = $url;
    }
    //http://www.youtube.com/watch?feature=player_embedded&v=m-t4pcO99gI
    //url is http://www.youtube.com/watch?v=xxxx
    else{
        parse_str($urls['query']);
        $yid = $v;
        if(!empty($feature)){
            $yid = end(explode('v=',$urls['query']));
            $arr = explode('&',$yid);
            $yid = $arr[0];
        }
    }
    if(!empty($yid)) {
    
    //return embed iframe
    if($return == 'embed'){
        return '<iframe data-position="'.($element_position).'" data-step="'.($element_step).'" width="'.($width?$width:540).'" height="'.($height?$height:300).'" src="http://www.youtube.com/embed/'.$yid.'?wmode=transparent?rel='.$rel.'" frameborder="0" ebkitAllowFullScreen mozallowfullscreen allowFullScreen></iframe>';
    }
    //return normal thumb
    else if($return == 'thumb' || $return == 'thumbmed'){
        return 'http://i1.ytimg.com/vi/'.$yid.'/default.jpg';
    }
    //return hqthumb
    else if($return == 'hqthumb' ){
        return 'http://i1.ytimg.com/vi/'.$yid.'/hqdefault.jpg';
    }
    // else return id
    else{
        return $yid;
    }
  }
  else if($vid) {
  $vimeoObject = json_decode(file_get_contents("http://vimeo.com/api/v2/video/".$vid.".json"));
   if (!empty($vimeoObject)) {
      //return embed iframe
      if($return == 'embed'){
      return '<iframe data-position="'.($element_position).'" data-step="'.($element_step).'" width="'.($width?$width:$vimeoObject[0]->width).'" height="'.($height?$height:$vimeoObject[0]->height).'" src="http://player.vimeo.com/video/'.$vid.'?title=0&byline=0&portrait=0" frameborder="0" webkitAllowFullScreen mozallowfullscreen allowFullScreen></iframe>';
    }
    //return normal thumb
    else if($return == 'thumb'){
      return $vimeoObject[0]->thumbnail_small;
    }
    //return medium thumb
    else if($return == 'thumbmed'){
      return $vimeoObject[0]->thumbnail_medium;
    }
    //return hqthumb
    else if($return == 'hqthumb'){
      return $vimeoObject[0]->thumbnail_large;
    }
    // else return id
    else{
      return $vid;
    }
   }
  }
}


/**
 * Remove permalink metabox (slug).
 *
 * @since	1.0.0
 */
function tus_remove_permalink_meta_box() {

    remove_meta_box( 'slugdiv', 'theux_slider', 'core' );
}
add_action( 'admin_menu', 'tus_remove_permalink_meta_box' );


/** 
 * Removing unused columns from the admin listing page.
 *
 * @since	1.0.0
 */
function remove_tus_columns($columns) {
    
    unset($columns['wps_post_thumbs']);
    return $columns;
}
add_filter('manage_theux_slider_posts_columns' , 'remove_tus_columns');


assets/js/public.js
;(function($, doc, win) {
  "use strict";
       
    $(window).load(function () {
        $('.slider').fractionSlider({
            'controls' : tusparams.controls, // controls on/off
            'pager' : tusparams.pager, // pager on/off
            'autoChange' : tusparams.autochange, // auto change slides
            'pauseOnHover' : tusparams.pauseonhover, // pauses slider on hover (current step will still be completed)
            'slideEndAnimation' : tusparams.slideendanimation, // if set true, objects will transition out before next slide moves in 
            'backgroundAnimation' : tusparams.backgroundanimation, // background animation
            'backgroundX' : tusparams.backgroundx, // background animation x distance
            'backgroundY' : tusparams.backgroundy, // background animation y distance
            'backgroundSpeed' : tusparams.backgroundspeed, // default background animation speed
            'backgroundEase' : tusparams.backgroundease, // default background animation easing
            'fullWidth': tusparams.fullwidth,
            'responsive': tusparams.responsive, // responsive slider (see below for some implementation tipps)
            'increase': tusparams.increase, // if set, slider is allowed to get bigger than basic dimensions
            'dimensions': tusparams.dimensions, // slider dimensions
        });
    });
})(jQuery, document, window);

theux-slider.php
<?php
/**
 * Plugin Name: TheUx Slider
 * Description: Premium open source slider.
 * Plugin URI: http://wordpress.org/plugins/theux-slider
 * Author: Daniel Zilli
 * Author URI: http://theux.co
 * License: GPLv2 (http://www.gnu.org/licenses/old-licenses/gpl-2.0.html)
 * Version: 0.9.9
 * Text Domain: tus
 * Domain Path: /assets/languages
 *
 * TheUx Slider is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 2 of the License, or
 * any later version.
 *
 * TheUx Slider is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * @package   TUS
 * @author    Daniel Zilli
 * @version   1.0.0
 */


/**
 * If this file is called directly, abort it.
 */
if ( ! defined( 'ABSPATH' ) ) exit;


/**
 * Load the helper library.
 */
require_once( plugin_dir_path( __FILE__ ) . 'includes/cuztom/cuztom.php');


/**
 * Administration.
 */
require_once( plugin_dir_path( __FILE__ ) . 'includes/admin.php' );


/**
 * Adds function to load slideshow within theme.
 */
function theux_slider( $slideshow = '' ) {
	require_once( plugin_dir_path( __FILE__ ) . 'includes/public.php' );
}


/**
 * Plugin class.
 */
if ( !class_exists('TheUx_Slider') ) {

	class TheUx_Slider {

		/**
		 * Plugin version, used for cache-busting of style and script file references.
		 */
		const VERSION = '1.0.0';


		/**
		 * Used as the text domain when internationalizing strings of text. It should
		 * match the Text Domain file header in the main plugin file.
		 */
		protected $plugin_slug = 'tus';


		/**
		 * Initialize the plugin by setting localization, filters, and shortcode functions.
		 */
		public function __construct() {

			/* Load plugin text domain. */
			add_action( 'init', array( $this, 'load_plugin_textdomain' ) );

			/* Load admin style sheet. */
			add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_admin_styles' ) );

			/* Load public style sheet and javacripts. */
			add_action( 'wp_enqueue_scripts', array( $this, 'enqueue_styles' ) );
			add_action( 'wp_enqueue_scripts', array( $this, 'enqueue_scripts' ) );

			/* Shortcode */
			add_shortcode( 'slider', array( $this, 'tus_shortcode' ) );
		}


		/**
		 * Load plugin text domain for translation.
		 */
		public function load_plugin_textdomain() {

			$domain = $this->plugin_slug;
			$locale = apply_filters( 'plugin_locale', get_locale(), $domain );

			load_textdomain( $domain, WP_LANG_DIR . '/' . $domain . '/' . $domain . '-' . $locale . '.mo' );
			load_plugin_textdomain( $domain, FALSE, basename( dirname( __FILE__ ) ) . '/assets/languages/' );
		}


		/**
		 * Enqueue admin-specific style sheet.
		 */
		public function enqueue_admin_styles($hook) {

			/* Only load the css on the necessary pages */
			if( $hook != 'edit.php' && $hook != 'edit-tags.php' && $hook != 'post-new.php' ) {
				return;
			}

			wp_enqueue_style( $this->plugin_slug .'-admin-styles', plugins_url( 'assets/css/admin.css', __FILE__ ), array(), self::VERSION );
		}
		

		/**
		 * Enqueue public style sheet.
		 */
		public function enqueue_styles() {
			wp_enqueue_style( $this->plugin_slug . '-plugin-styles', plugins_url( 'assets/css/public.css', __FILE__ ), array(), self::VERSION );
		}


		/**
		 * Enqueue and register public JavaScript files.
		 */
		public function enqueue_scripts() {
			wp_enqueue_script( $this->plugin_slug . '-slider-script', plugins_url( 'assets/js/jquery.fractionslider.min.js', __FILE__ ), array( 'jquery' ), '0.9.9.9');
			wp_register_script ( $this->plugin_slug . '-public-script', plugins_url( 'assets/js/public.js', __FILE__ ), array( 'tus-slider-script' ), self::VERSION, TRUE);
		}


		/**
		 * Define the shortcode and its respective default values.
		 */
		public function tus_shortcode($atts) {

			$data = shortcode_atts ( 
				array(
					'slideshow' 			=> '',
					'controls' 				=> TRUE,
					'pager'					=> FALSE,
					'autochange'			=> FALSE,					
					'pauseonhover'			=> FALSE,
					'slideendanimation' 	=> FALSE,
		   			'backgroundanimation'	=> FALSE, 
		            'backgroundx' 			=> '500', 
		            'backgroundy' 			=> '500', 
		            'backgroundspeed' 		=> '2500', 
		            'backgroundease'		=> 'easeOutCubic',
					'fullwidth' 			=> FALSE,
					'responsive' 			=> TRUE,
					'dimensions' 			=> '',
					'increase'				=> FALSE,
				), 
				$atts 
			);
			
			$slideshow_att = $data['slideshow'];
			ob_start();
			theux_slider( $slideshow = $slideshow_att );	
			$theux_slider_content = ob_get_clean();
			
			wp_enqueue_script ( $this->plugin_slug . '-public-script');
			wp_localize_script( $this->plugin_slug . '-public-script', 'tusparams', $data );

			return $theux_slider_content;
		}

	} // End of TheUx_Slider class.


	/**
	 * Initialise the plugin.
	 */
	new TheUx_Slider();

} // End if class_exists check.
